ansible_ai_connect/ai/api/telemetry/schema1.py

```diff
@@ -0,0 +1,326 @@
+        self.rh_user_org_id = user.org_id
+        self.groups = list(user.groups.values_list("name", flat=True))
+
+    def set_exception(self, exception):
+        if not exception:
+            return
+        self.exception = True
+        self.response.exception = str(exception)
+        self.problem = (
+            exception.problem
+            if isinstance(exception, MarkedYAMLError)
+            else str(exception) if str(exception) else exception.__class__.__name__
+        )
+
+    def set_request(self, request):
+        pass
+
+    def set_response(self, response):
+        def get_message(response):
+            if response.status_code < 400:
+                return ""
+            full_content = str(getattr(response, "content", ""))
+            if len(full_content) > 200:
+                return full_content[:200] + "…"
+            else:
+                return full_content
+
+        self.response.status_code = response.status_code
+        if response.status_code >= 400:
+            self.response.error_type = getattr(response, "error_type", None)
+            self.response.message = get_message(response)
+            self.response.status_text = (getattr(response, "status_text", None),)
+
+    def set_validated_data(self, validated_data):
+        for field_name, value in validated_data.items():
+            if hasattr(self, field_name):
+                setattr(self, field_name, value)
+
+        # TODO: improve the way we define the model in the payload.
+        try:
+            model_mesh_client = apps.get_app_config("ai").model_mesh_client
+            self.modelName = model_mesh_client.get_model_id(
+                self.rh_user_org_id, str(validated_data.get("model", ""))
+            )
+            print(f"self.modelName={self.rh_user_org_id}")
+        except (WcaNoDefaultModelId, WcaModelIdNotFound, WcaSecretManagerError):
+            logger.debug(
+                f"Failed to retrieve Model Name for Feedback.\n "
+                f"Org ID: {self.rh_user_org_id}, "
+                f"User has seat: {self.rh_user_has_seat}, "
+                f"has subscription: {self.user.rh_org_has_subscription}.\n"
+            )
+
+    @classmethod
+    def init(cls, user, validated_data):
+        print("init()")
+        schema1_event = cls()
+        schema1_event.set_user(user)
+        schema1_event.set_validated_data(validated_data)
+        return schema1_event
+
+    def as_dict(self):
+        # NOTE: The allowed fields should be moved in the event class itslef
+        def my_filter(a, v):
+            return a.name not in ["event_name", "user"]
+
+        return asdict(self, filter=my_filter, recurse=True)
+
+
+@define
+class CompletionRequestPayload:
+    context: str = field(validator=validators.instance_of(str), converter=str, default="")
+    prompt: str = field(validator=validators.instance_of(str), converter=str, default="")
+
+
+@define
+class CompletionEvent(Schema1Event):
```

Comment:
Where is this event used?

Reply:
It's associated with the Completion view here: https://github.com/ansible/ansible-ai-connect-service/pull/1147/files#diff-ecfb6919dfd8379aafba96af7457b253e4dce528897dfe6bfc207ca2b3b2ada9R221